### ISSUE TITLE
npm doesn't use sinopia registry

### ISSUE THREAD

**Issue by mmedvedec** [rlidwka/sinopia#371]
Hi all,

After setting npm to use sinopia registry as my local cache,
It seems that npm is still querying _registry.npmjs.org_ for packages.

```
$ npm config get registry
http://192.168.1.100:4873/
```

```
$ npm install -g m3u

 info  <-- 192.168.1.100 requested 'GET /m3u'
 info  --> making request: 'GET https://registry.npmjs.org/m3u'
 http  --> 304, req: 'GET https://registry.npmjs.org/m3u', bytes: 0/0
 http  <-- 304, user: os7een, req: 'GET /m3u', bytes: 0/0
 info  <-- 192.168.1.100 requested 'GET /chunked-stream'
 info  --> making request: 'GET https://registry.npmjs.org/chunked-stream'
 http  --> 304, req: 'GET https://registry.npmjs.org/chunked-stream', bytes: 0/0
 http  <-- 304, user: os7een, req: 'GET /chunked-stream', bytes: 0/0
```

And under WebUI I also can't see that package,
But it's locally stored inside my package directory.

```
storage: /home/sinopia/packages

$ ls -al /home/sinopia/packages/m3u
total 20
drwxr-xr-x 2 sinopia sinopia 4096 Feb  5 12:18 .
drwxr-xr-x 7 sinopia sinopia 4096 Feb  5 12:18 ..
-rw-r--r-- 1 sinopia sinopia 4513 Feb  5 11:38 m3u-0.0.2.tgz
-rw-r--r-- 1 sinopia sinopia 2512 Feb  5 12:18 package.json
```

Thanks for your clarification,
Best regards.

**Comment by excenter** [comment]
Here's my experience using sinopia: 
When you request a package, it will immediately look at npmjs.org and see if the cached version is the latest. If it is the latest, or you've overwritten the public one by versioning yours well above what they are versioned at, OR you are requesting a private module, it will deliver the cached version. 
TL;DR 
It will call to npmjs.org at every install request, delivering the "latest" version based on what it has cached, or can see at npmjs.lrg.
